### PR TITLE
Support 2.1.53 desktop version

### PR DESCRIPTION
**Pull request by dae**
Posting this for reference; recommend you pull it into an ankidroid-2.1.53 branch instead of main, which a new -Backend PR can then consume.

This reworks David's previous AnkiDroid-specific changes to work with the latest desktop code, and moves most of the additional Rust code from the -Backend repo into a patch on rslib. This greatly cuts down on the JNI boilerplate, and improves error handling. Most of the additional code is in separate files, so conflicts when rebasing over newer Anki versions should be fairly rare.

### CODE REVIEW

**Comment by dae** [comment]
Please hold off on merging this until I have a chance to fix a Windows/Mac issue with the protobuf files.

**Comment by dae** [comment]
Should be good to go now; all tests are passing in the -Backend repo.

**Comment by mikehardy** [comment]
Just agreeing with the mechanical / source integration comment here, preliminarily:

> recommend you pull it into an ankidroid-2.1.53 branch instead of main, which a new -Backend PR can then consume.

The rest seems like deciding where to move the dirt under the rug, and I have a default position of agreeing with you as rust lib owner and following "implementor's choice" bias, but I want to review to make sure of course

**Review by mikehardy — APPROVED** [review]
It's my sincere hope to have as much of this as possible upstream-ed but this is already an improvement structurally as a sort of git branch-extension off the upstream tag. Not too hard to follow

I've done a branch here on our fork at your upstream 2.1.53 tag, re-targeted this PR to point to it and that should be how to do it unless my git-fu fails me
